### ISSUE TITLE
Support for missing values in integer types

### ISSUE THREAD

**Issue by fehtemam** [noirello/pyorc#33]
Does the writer support writing null values to integer types? I know ORC supports null in all types but I can't seem to figure out how to make it work with the writer. A reproducible example is something like this:

```python
import random
import pandas as pd
# create empty dataframe
orc_sample = pd.DataFrame()
# make an int8 column (equivalent to tinyint)
orc_sample['int8_col'] = [random.randrange(0, 128, 2) for i in range(0, 10)]
# replicate that and add a missing value to create a second column
orc_sample['Int8_col'] = orc_sample['int8_col'].astype('Int8')
orc_sample.loc[6, 'Int8_col'] = None
# then write
strct = "struct<" + "int8_col:tinyint,Int_8:tinyint" + ">"
bytefile = io.BytesIO()
with pyorc.Writer(bytefile,
                  strct,
                  struct_repr=pyorc.StructRepr.DICT,
                  compression=pyorc.CompressionKind.ZLIB) as writer:
    writer.writerows(orc.to_dict(orient="records"))
```   
I keep getting: Item <NA> cannot be cast to long int at struct field name 'Int8_col'
I'm wondering if anyone has successfully written integer types with missing values?! Or if there is any trick to make it work?!

**Comment by noirello** [comment]
I think when you assign `None` to the column it actually fill it with `numpy.nan`. That's something that the Writer cannot cast. You have to replace it with `None`, before passing it to the Writer.

**Comment by fehtemam** [comment]
Thanks so much for the prompt response! Yes, you are right. I looked more into it and realized that there was no way for ORC library to understand pd.NA. pd.NA is Pandas nullable int type (that's how None in my code above gets represented) which was introduce I think two years ago to fill in the gap in Python and Numpy since neither of them support null in integer types. So I had to first convert Pandas dataframe to a dictionary and then replace all pd.NA with None and then after that the writer wrote them perfectly. Thanks again for this package! It's awesome!
